src/libraries/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -637,15 +637,10 @@ public bool IsPowerOfTwo
 
                 if (_sign != 1)
                     return false;
+
                 int iu = _bits.Length - 1;
-                if (!BitOperations.IsPow2(_bits[iu]))
-                    return false;
-                while (--iu >= 0)
-                {
-                    if (_bits[iu] != 0)
-                        return false;
-                }
-                return true;
+
+                return BitOperations.IsPow2(_bits[iu]) && _bits.AsSpan().IndexOfAnyExcept(0u) == iu;
```

Comment:
```suggestion
                return BitOperations.IsPow2(_bits[iu]) && !_bits.AsSpan(0, iu).ContainsOfAnyExcept(0u);
```

Can it be better with `ContainsOfAnyExcept`?

Reply:
`ContainsAnyExcept` currently calls to `IndexOfAnyExcept`, but if anything it is marginally better (no worse), and probably shows clearer intent. Thanks